docs/integrations/azure/locations.md

```diff
@@ -41,24 +41,53 @@ Using a personal access token (PAT):
 integrations:
   azure:
     - host: dev.azure.com
-      token: ${AZURE_TOKEN}
+      credentials:
+        - personalAccessToken: ${PERSONAL_ACCESS_TOKEN}
```

Comment:
Without an org this didn't work for me. That either needs to be added or maybe something needs to be fixed

Reply:
@awanlin I tried this myself, and for me, it does work. Can you elaborate a bit on what problem you are facing? Do you see any error messages?

Also, what does your config look like? If you, for example, have two `credentials` without an org, the first one listed will be used. In the case below, a `ManagedIdentityCredential` would be used because it's the first `credential` without an org:
```
- host: dev.azure.com
  credentials:
    - clientId: ....
    - personalAccessToken: ....
```

Even if this isn't the problem in your case, I can see how this could confuse the users. So I added two additional checks:
- reject the config when more than one credential without any `organizations` is specified
- reject the config when multiple credentials specify the same organization

p.s.: I want to get this into `1.1.17`, so can you poke one of the other maintainers to have a look as well? I mentioned @Rugvip a few times now, I mentioned the other @backstage/catalog-maintainers, and I dropped this PR in the `catalog` channel on Discord. However, I still need another review from a maintainer to get this merged.

Reply:
There was no error it simply did not pull in the entity from my Azure DevOps Server instance but did for my Azure DevOps Cloud. I'll re-test today but as I mentioned I'm very worried about this going out in today's release. 

This is how my config was setup:

```yaml
azure:
    - host: azure-devop-server.company.com
      credentials:
      - personalAccessToken: ${TOKEN}
    - host: dev.azure.com
      credentials:
      - personalAccessToken: ${ANOTHER_TOKEN}
        organisations:
        - some-org
```

Reply:
Ok, re-tested and I found the issue: when you create a PAT that is only for a specific org but don't include the org in the config like I did above it doesn't work. Before I wasn't getting any error but now I get a 401 which at least forced me to play around with this. The fix was that I had to create the PAT using the "For all accessible organizations" option. This same PAT (setup for 1 org) worked before in my prior testing .

Reply:
When creating a PAT it always seems to default to one of the Orgs you have access to and not the "For all accessible organizations" option. At least that's what I get when I test with Server and Cloud, can you confirm? Hopefully I'm wrong. Based on that most adopters will have an Org specific PAT in place that they know is good, try to use it and run into this issue if they don't include the Org.

Reply:
Interesting 🤔 , I can't easily explain that based on the changes I made to support multi-org personal access tokens. I'm pretty sure that the PAT I'm using to test is in fact for a single organization, but I'll test both scenarios to see if I get the same error.

Reply:
Ok, did more testing. My main focus is being able to pull from both Server and Cloud instances. In my testing that has been the setup I've tested the most. If I just use:

```yaml
integrations:
  azure:
    - host: sever.company.com
      credentials:
        - personalAccessToken: ${ORG_SPECIFIC_TOKEN}
``` 

That works fine and pulls in the entity I have setup. But as soon as I add in this setup I run into the issue I posted about earlier:

```
azure:
    - host: azure-devop-server.company.com
      credentials:
      - personalAccessToken: ${ORG_SPECIFIC_TOKEN}
    - host: dev.azure.com
      credentials:
      - personalAccessToken: ${ANOTHER_TOKEN}
        organisations:
        - some-org
```
For the above to work I need to add the org or use an org-wide PAT

Reply:
That is weird, the organization is only used for matching the URLs and when no credential for a specific organization is found then I try to find a credential without an organization.

I do not treat the PAT differently so I wonder why this doesn’t work for DevOps server when you use an org specific PAT. Tomorrow I have a day off, so I’ll look into it further on Thursday.

Reply:
@awanlin if I understand correctly it works when you specify only 1 azure integration but when you add a 2nd integration, for dev.azure.com in this case, it doesn’t work anymore and you have to specify the org for DevOps Server?

Reply:
Yes, that's exactly what I'm running in to. I have not tested the reverse but can do that later today

Reply:
@awanlin I tried to replicate the scenario with your config, but it is impossible to specify an `organization` for an Azure DevOps server. You should get the following error:

So, modifying your configuration like this:

```diff
integrations:
  azure:
    - host: azure-devops-server.company.com
      credentials:
        - personalAccessToken: `${ORG_SPECIFIC_TOKEN}`
+         organizations:
+           - my-org
    - host: dev.azure.com
      credentials:
        - personalAccessToken: ${ANOTHER_TOKEN}
```

Results in this error message:
```text
Backend failed to start up Error: Invalid Azure integration config for azure-devops-server.company.com, you cannot specify organizations for hosts other than dev.azure.com
```

The backend only starts when I remove the `organizations` property again:
```diff
integrations:
  azure:
    - host: azure-devops-server.company.com
      credentials:
        - personalAccessToken: `${ORG_SPECIFIC_TOKEN}`
-         organizations:
-           - my-org
    - host: dev.azure.com
      credentials:
        - personalAccessToken: ${ANOTHER_TOKEN}
```

I don't have access to an Azure DevOps server, so I cannot test whether the `${ORG_SPECIFIC_TOKEN}` works for me, but the `dev.azure.com` integration works with an org-specific token without specifying the `organizations` field.

Did you mean you must specify the `organizations` field on the credential for `dev.azure.com` to make it work?
```diff
integrations:
  azure:
    - host: azure-devops-server.company.com
      credentials:
        - personalAccessToken: `${ORG_SPECIFIC_TOKEN}`
    - host: dev.azure.com
      credentials:
        - personalAccessToken: ${ANOTHER_ORG_SPECIFIC_TOKEN}
+         organizations:
+           - my-org
```

Also, I did notice that you used the British English form (organi**s**ations) in your example, whereas the config and code behind use the American English form (organi**z**ations). But I assume this is a typo in your GitHub comment instead of a typo in your configuration. It makes no sense that specifying the organi**s**ations field on a credential causes it to work.

Reply:
@sanderaernouts just a heads up I can't re-test until the afternoon, I have meeting all morning. Also, would it make sense for us to move this to Discord? My user name is `Ahhhndre`. I'm also thinking that we should try get on a call as well that way you can get all the information you need

Reply:
I know we are meeting shortly but just wanted to comment on the spelling, I have been using `organisations` in all my testing and never ran into any errors with this so that makes this all the more odd.

Reply:
@awanlin based on our meeting earlier today, I made the following changes:
- support `organizations` for Azure DevOps server (including servers prefixed with `/tfs`)
- updated the schema in `config.d.ts` to reflect the new Azure integration configuration

Reply:
Awesome, will test now!

Reply:
Testing is all good 🚀